rest/oidc_api.go

```diff
@@ -118,6 +149,30 @@ func (h *handler) handleOIDCCallback() error {
 		return base.HTTPErrorf(http.StatusBadRequest, "Unable to identify provider for callback request")
 	}
 
+	// Validate state parameter to prevent cross-site request forgery (CSRF) when DisableCallbackState is not enabled.
+	if provider.DisableCallbackState != nil && !*provider.DisableCallbackState {
+		stateCookie, err := h.rq.Cookie(stateCookieName)
+
+		if err != nil && err != http.ErrNoCookie {
+			base.Warnf("Unexpected error returned from request.Cookie(%s): %v", stateCookieName, err)
+			return ErrReadStateCookie
+		}
+
+		if err != nil {
+			return ErrReadStateCookie
+		}
+		if stateCookie == nil {
+			return ErrNoStateCookie
+		}
```

Comment:
I think this always needs to be returning `ErrNoStateCookie` now we've done the explicit error check above.

```suggestion
		if err != nil || stateCookie == nil {
			return ErrNoStateCookie
		}
```

Reply:
Yeah, this makes sense. Committed this in latest revision.